src/odb/src/db/dbNet.cpp

```diff
@@ -95,14 +96,15 @@ _dbNet::_dbNet(_dbDatabase*, const _dbNet& n)
       _ccAdjustOrder(n._ccAdjustOrder)
 
 {
+  guide_tbl_ = new dbTable<_dbGuide>(db, this, *n.guide_tbl_);
```

Comment:
A table per net is going to be costly in terms of memory.  The table is designed for a lot of objects and allocates in chunks while a single net may not have many guides.

The table should be on the block and the guide can have prev/next-in-net id field for a linked list of guides per net.  The net would have a head guide id.  Similar to bterms on the net.